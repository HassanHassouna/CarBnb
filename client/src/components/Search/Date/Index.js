import React, { forwardRef } from "react"
import TextField from "@mui/material/TextField"
import { AdapterDateFns } from "@mui/x-date-pickers/AdapterDateFns"
import { LocalizationProvider } from "@mui/x-date-pickers/LocalizationProvider"
import { DatePicker } from "@mui/x-date-pickers/DatePicker"
import { makeStyles } from "@material-ui/core/styles"

const Datee = forwardRef((props, ref) => {
  const useStyles = makeStyles(() => ({
    noBorder: {
      border: "none",
    },
  }))
  const [value, setValue] = React.useState(null)
  const classes = useStyles()
  return (
    <LocalizationProvider dateAdapter={AdapterDateFns}>
      <DatePicker
        toolbarPlaceholder={props.label}
        value={
          value === null
            ? props.label === "From"
              ? new Date()
              : new Date().setDate(new Date().getDate() + 3)
            : value
        }
        inputRef={ref}
        InputProps={{
          classes: {
            notchedOutline: props.classes ? props.classes : classes.noBorder,
          },
        }}
        onChange={(newValue) => {
          if (window.location.href !== "http://localhost:3000/") {
            setValue(newValue)
            props.label === "From"
              ? props.setDataFrom(newValue)
              : props.setDataUntil(newValue)
          } else {
            setValue(newValue)
          }
        }}
        renderInput={(params) => <TextField sx={{minWidth:145}} {...params} />}
      />
    </LocalizationProvider>
  )
})

export default Datee
