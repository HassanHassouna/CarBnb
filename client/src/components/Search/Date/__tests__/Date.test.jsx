import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import Date from '../Index';

describe('Date', () => {
  test('should render with no parameters', () => {
    const { date } = render(<Date />);

    expect(date).toMatchSnapshot();
  });
});
