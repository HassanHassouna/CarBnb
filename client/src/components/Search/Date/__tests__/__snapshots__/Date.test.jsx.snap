// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Date should render with no parameters 1`] = `undefined`;
